python/core/auto_generated/qgsdataitemprovider.sip.in

```diff
@@ -38,7 +38,7 @@ or directory being inspected, parent item is a valid QgsDirectoryItem
 Human-readable name of the provider name
 %End
 
-    virtual int capabilities() = 0;
+    virtual int capabilities() const = 0;
```

Comment:
@3nids 

if I understand it correctly you cannot forward declare enum (since the return value is `QgsDataProvider::DataCapability`)
and we do not want to add additional include everywhere (all dataitemproviders to include `qgsdataprovider.h`)

so 1) Move  DataCapability to QgsDataItemProvider ?
or 2) Move DataCapability outside QgsDataProvider to enum class (similarly to QgsMapLayersType) `QgsDataCapability`
3.)  Move it to Qgis.h

Reply:
Yes, 3 options work.
Regarding 3, I am not sure but this might require to merge #9734 first, but not sure.
We just need to define before next release what we prefer between 2 and 3 and stick to this.

But in any case, you should use an enum class, embedded or not. This produces nicer API docs and will be the norm in PyQt 6.

Reply:
created 3) `Qgis::DataCapabilities` QFlag

Reply:
is this back to int?

Reply:
Yes I have reverted it back to int... Unfortunately I was getting stuck again and again with sip with the newly introduced Qgis::DataCapabilities and was unable to fix resulting API breaks. And given the brutal amount of code it made sense to me to keep that method as it was and leave it for QGIS 4